src/posix/platform/netif.cpp

```diff
@@ -680,11 +681,18 @@ template <size_t N> otError AddRoute(const uint8_t (&aAddress)[N], uint8_t aPref
     AddRtAttrUint32(&req.header, sizeof(req), RTA_PRIORITY, aPriority);
     AddRtAttrUint32(&req.header, sizeof(req), RTA_OIF, netifIdx);
 
+    inet_ntop(req.msg.rtm_family, aAddress, addrStrBuf, sizeof(addrStrBuf));
```

Comment:
Since we are using `inet_ntop` how about using its defined string size constant (`INET6_ADDRSTRLEN`) instead of OT constant `OT_IP6_ADDRESS_STRING_SIZE`?

 (`OT_IP6_ADDRESS_STRING_SIZE` is safe for use along with` otIp6AddressToString()`)